sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -776,6 +774,35 @@ abstract class TypeChecks[RET] {
     }.mkString(", ")
   }
 
+  /** 
+   * Original log does not print enough info when timezone is not UTC, 
+   * here check again to add UTC info and return other unsupported types.
+   */
+  private def tagTimestampTypeAndReturnOthers(
+    unsupportedTypes: Map[DataType, Set[String]],
+    meta: RapidsMeta[_, _, _]
+    ): Map[DataType, Set[String]] = {
+    def checkTimestampType(dataType: DataType): Boolean = dataType match {
+        case TimestampType if !TypeChecks.areTimestampsSupported() => {
+          meta.willNotWorkOnGpu(s"your timezone isn't in UTC (JVM:" +
+            s" ${ZoneId.systemDefault()}, session: ${SQLConf.get.sessionLocalTimeZone})." +
+            s" Set both of the timezones to UTC to enable TimestampType support")
+          false
+        }
+        case ArrayType(elementType, _) =>
```

Comment:
If one column is `array(timestamp)` , `ArrayType` is unsupported, `TimestampType` is supported and time zone is not UTC, this method returns false, `tagTimestampTypeAndReturnOthers` will remove this `array(timestamp)` type.
Actually, we can't distinguish the unsupported reasons for nested types.
Let's just add one extra error msg of UTC and remain the original error msg.

Reply:
Nice catch, fixed.